mlflow/deployments/databricks/__init__.py

```diff
@@ -1,39 +1,123 @@
+import posixpath
+from typing import Any, Dict, Optional
+
 from mlflow.deployments import BaseDeploymentClient
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.rest_utils import augmented_raise_for_status, http_request
 
 
 class DatabricksDeploymentClient(BaseDeploymentClient):
     def create_deployment(self, name, model_uri, flavor=None, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def update_deployment(self, name, model_uri=None, flavor=None, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def delete_deployment(self, name, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def list_deployments(self, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def get_deployment(self, name, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
+    def _call_endpoint(
+        self,
+        *,
+        method: str,
+        prefix: str = "/api/2.0",
+        route: Optional[str] = None,
+        json_body: Optional[Dict[str, Any]] = None,
+    ):
+        call_kwargs = {}
+        if method.lower() == "get":
+            call_kwargs["params"] = json_body
+        else:
+            call_kwargs["json"] = json_body
+
+        response = http_request(
+            host_creds=get_databricks_host_creds(self.target_uri),
+            endpoint=posixpath.join(prefix, "serving-endpoints", route if route else ""),
+            method=method,
+            timeout=300,
+            raise_on_status=False,
+            **call_kwargs,
+        )
+        augmented_raise_for_status(response)
+        return response.json()
+
     def predict(self, deployment_name=None, inputs=None, endpoint=None):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        return self._call_endpoint(
+            method="POST",
+            prefix="/",
+            route=posixpath.join(endpoint, "invocations"),
+            json_body=inputs,
+        )
 
     def create_endpoint(self, name, config=None):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        config = config.copy() if config else {}
+        payload = {"name": name, "config": config}
+        if task := config.pop("task", None):
+            payload["task"] = task
+        return self._call_endpoint(method="POST", json_body=payload)
```

Comment:
We need to support `tags`?

https://docs.databricks.com/api/workspace/servingendpoints/create

Reply:
Great catch - yes :)

Reply:
updated